verifier/helpers.go

```diff
@@ -183,3 +189,195 @@ func isTSATrustStoreInPolicy(policyName string, trustStores []string) (bool, err
 	}
 	return false, nil
 }
+
+// verifyTimestamp provides core verification logic of authentic timestamp under
+// signing scheme `notary.x509`.
+func verifyTimestamp(ctx context.Context, policyName string, trustStores []string, signatureVerification trustpolicy.SignatureVerification, x509TrustStore truststore.X509TrustStore, r revocation.Revocation, outcome *notation.VerificationOutcome) *notation.ValidationResult {
+	logger := log.GetLogger(ctx)
+
+	signerInfo := outcome.EnvelopeContent.SignerInfo
+	performTimestampVerification := true
+	// check if tsa trust store is configured in trust policy
+	tsaEnabled, err := isTSATrustStoreInPolicy(policyName, trustStores)
+	if err != nil {
+		return &notation.ValidationResult{
+			Error:  fmt.Errorf("failed to check tsa trust store configuration in turst policy with error: %w", err),
+			Type:   trustpolicy.TypeAuthenticTimestamp,
+			Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeAuthenticTimestamp],
+		}
```

Comment:
This error wrapping seems redundant in `verifyTimestamp()`.

How about
```go
func verifyTimestamp(...) error
```
and in `verifyAuthenticTimestamp()` do something like
```go
return &notation.ValidationResult{
	Error:  verifyTimestamp(...);,
	Type:   trustpolicy.TypeAuthenticTimestamp,
	Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeAuthenticTimestamp],
}
```

Reply:
Yeah, good idea. Updating.